README.md

```diff
@@ -28,54 +28,47 @@ Require the library
 require 'graphql/batch'
 ```
 
-Define a GraphQL::Batch::Query derived class. Use group_key to specify which queries can be reduced into a batch query, and an execute class method which makes that batch query and passes the result to each individual query.
+Define a custom loader, which is initialized with arguments that are used for grouping and an execute method for doing the batch query. The load method can be overriden to coerce the type of the key.
 
 ```ruby
-class FindQuery < GraphQL::Batch::Query
-  attr_reader :model, :id
+class RecordLoader < GraphQL::Batch::Loader
+  attr_reader :model
 
-  def initialize(model, id, &block)
+  def initialize(model)
+    super()
     @model = model
-    @id = id
-    super(&block)
   end
 
-  # super returns the class name
-  def group_key
-    "#{super}:#{model.name}"
+  def load(key)
```

Comment:
I wouldn't bother documenting this, I imagine it will be unnecessary 99% of the time and doesn't do anything that isn't obvious with usual ruby method overrides?